include/rocksdb/utilities/env_librados.h

```diff
@@ -0,0 +1,186 @@
+
+  // Delete the named file.
+  rocksdb::Status DeleteFile(const std::string& fname);
+
+  // Create the specified directory. Returns error if directory exists.
+  rocksdb::Status CreateDir(const std::string& dirname);
+
+  // Creates directory if missing. Return Ok if it exists, or successful in
+  // Creating.
+  rocksdb::Status CreateDirIfMissing(const std::string& dirname);
+
+  // Delete the specified directory.
+  rocksdb::Status DeleteDir(const std::string& dirname);
+
+  // Store the size of fname in *file_size.
+  rocksdb::Status GetFileSize(const std::string& fname, uint64_t* file_size);
+
+  // Store the last modification time of fname in *file_mtime.
+  rocksdb::Status GetFileModificationTime(const std::string& fname,
+                                         uint64_t* file_mtime);
+  // Rename file src to target.
+  rocksdb::Status RenameFile(const std::string& src,
+                            const std::string& target);
+  // Hard Link file src to target.
+  rocksdb::Status LinkFile(const std::string& src, const std::string& target);
+
+  // Lock the specified file.  Used to prevent concurrent access to
+  // the same db by multiple processes.  On failure, stores nullptr in
+  // *lock and returns non-OK.
+  //
+  // On success, stores a pointer to the object that represents the
+  // acquired lock in *lock and returns OK.  The caller should call
+  // UnlockFile(*lock) to release the lock.  If the process exits,
+  // the lock will be automatically released.
+  //
+  // If somebody else already holds the lock, finishes immediately
+  // with a failure.  I.e., this call does not wait for existing locks
+  // to go away.
+  //
+  // May create the named file if it does not already exist.
+  rocksdb::Status LockFile(const std::string& fname, rocksdb::FileLock** lock);
+
+  // Release the lock acquired by a previous successful call to LockFile.
+  // REQUIRES: lock was returned by a successful LockFile() call
+  // REQUIRES: lock has not already been unlocked.
+  rocksdb::Status UnlockFile(rocksdb::FileLock* lock);
+
+  // *path is set to a temporary directory that can be used for testing. It may
+  // or many not have just been created. The directory may or may not differ
+  // between runs of the same process, but subsequent calls will return the
+  // same directory.
+  rocksdb::Status GetTestDirectory(std::string* path);
+
+  // Create and return a log file for storing informational messages.
+  rocksdb::Status NewLogger(
+    const std::string& fname,
+    std::shared_ptr<rocksdb::Logger>* result);
+
+  // Get full directory name for this db.
+  rocksdb::Status GetAbsolutePath(const std::string& db_path,
+      std::string* output_path);
+
+  // Generate unique id
+  std::string GenerateUniqueId();
+  explicit EnvLibrados(const std::string& fsname, const std::string& config_path);
+  ~EnvLibrados() {
+    _rados.shutdown();    
+  } 
+private:
+  std::string _fsname;
+  std::string _db_id;
```

Comment:
These are confusing.  I think this code is confusing the create and open existing cases.  The constructor sets _db_id = fsname + random id in teh constructore.. I think which means you can't open an existing database.

I would simplify this to just take db_id in the ctor and drop fsname (I'm not sure what the difference is?).